tools/targets/PSOC6.py

```diff
@@ -16,17 +16,22 @@
 #
 
 import os
+import sys
+
 import platform
 import subprocess
 import errno
 from array import array
 from struct import (pack, unpack)
-from shutil import copyfile
-from intelhex import IntelHex
+from shutil import copy2
+from intelhex import IntelHex, hex2bin, bin2hex
 from intelhex.compat import asbytes
 
 from ..config import ConfigException
 
+from pathlib import Path, PurePath
```

Comment:
In previous discussions, moving to Python 3 was acceptable since the Mbed OS CI had planned to move to Python 3 anyway.  Lets discuss this further.

Reply:
@madchutney Unfortunately, our post build hook for adding signature is only working on Python 3, we had previously discussed above, that it acceptable.

Reply:
@maclobdell @romanjoe We already make best efforts to support Python 3 but this is not the same as dropping support for Python 2. I believe this is planned for some time after Python 2 sunsets and needs to be communicated to our customers and partners.

When this happens it must be a product decision. Not only does it impact our users, for example forcing Mac users to install Python 3, it also means that our CI systems, many of which explicitly check for Python 2 compatibility will be broken. This may take significant effort and we need to engage with all impacted teams.

Reply:
Just checked the road map, full Python 3 support should be in place for Mbed OS 5.15. Then there will be a grace period for us to allow our users to transition if they haven't already.

Reply:
@madchutney this tool is specific for Cypress target, and Cypress users of this target are required to use python 3 (for tools outside of mbed-os also). What would be the purpose of adding python 2 support in this case, as PSoC64 users will not be able to use it in their environment overall? 
This does not place any demands on the mbed-os platform itself to be python 3 only.

Reply:
As we are almost compatible with python3 - current Cypress users should not have a problem. However, our users will have. Same is for CI. we got jobs running py2.

Please test this PR with py2 as well. I've just received a report from CI test job for this PR about failing with py2.

Reply:
@csd-cypress You have referred to the PyPI package not the built in library that was introduced in Python 3.4 https://docs.python.org/3/library/pathlib.html

Reply:
@madchutney pathlib is included in root requirements.txt and from last commit for Python 2.7 sign_image function will not be called at all. so now compilation is possible on Python 2.7

Reply:
I didn't see the change to requirements.txt, as long as it works with python 3.5 and later the package is PyPI is only declared up to 3.4.

Reply:
this is from https://pypi.org/project/pathlib/ module

> Python 3.2 or later is recommended, but pathlib is also usable with Python 2.7 and 2.6.

Python 3.4+ has this module included as built in library.

these modules where added to root requrements.txt in scope of this PR
`cryptography>=2.4.x,<2.5`
`Click>=7.0,<7.1`
`pathlib>=1.0.1`

Reply:
I understand, my question was whether or not the built in library (>= Python 3.4) would work ok with the unconditional import included in requirements.txt

Reply:
@madchutney was not sure about that and decided to delete this requirement at all and make import of pathlib inside functions, which are called only when build is happening under Python 3. Please see latest commit on this PR.